lib/package.js

```diff
@@ -471,9 +471,21 @@ exports.download = function(pkg, options, installDeps) {
 
         // get the override
         if (endpoint.getOverride) {
-          if (!override && config.pjson.overrides[pkg.exactName])
-            ui.log('warn', 'Using local override for `' + pkg.exactName + '`');
-          return endpoint.getOverride(pkg.registry, pkg.package, pkg.version, override || config.pjson.overrides[pkg.exactName]);
+          if (!override) {
+            for (var overrideName in config.pjson.overrides) {
+              if (overrideName.indexOf(pkg.name) !== -1) {
+                var overrideVersion = overrideName.split('@')[1];
+
+                // use an override if it is compatible per semver
+                if (semver.match(overrideVersion, pkg.version)) {
```

Comment:
I was thinking mostly about the local overrides. For me it would be convenient if I could use any semver pattern in the local override (I am not sure about a use case). I tested on this override:

```
"overrides": {
      "github:components/ember@^1.13.2": {
        "main": "ember.debug",
        "files": [
          "ember.debug.js",
          "ember.prod.js"
        ],
        "dependencies": {
          "jquery": "github:components/jquery@^2.0.3"
        }
      }
    }
```